js/plates-model/plate.js

```diff
@@ -79,6 +84,41 @@ export default class Plate extends PlateBase {
     return totalTorque.applyMatrix3(this.invMomentOfInertia)
   }
 
+  updateCenter () {
+    if (this.center) {
+      const centerField = this.fieldAtAbsolutePos(this.center)
+      // Don't bother moving the label if it's still on a visible field
+      if (centerField && !centerField.colliding) {
+        return
+      }
+    }
+
+    // If the current label position is no longer valid, move it back to the center
+    let sum = new THREE.Vector3(0, 0, 0)
+    this.fields.forEach(field => {
+      if (!field.colliding) {
+        sum = sum.add(field.absolutePos)
+      }
+    })
+
+    // Resize to the surface of the sphere
+    const center = sum.normalize()
+
+    let closestPoint = new THREE.Vector3(0, 0, 0)
+    let minDist = Number.MAX_VALUE
+    this.fields.forEach(field => {
+      // Ensure we label a visible field
+      if (!field.colliding) {
+        const dist = field.absolutePos.distanceTo(center)
+        if (dist < minDist) {
+          closestPoint = field.absolutePos
+          minDist = dist
+        }
+      }
+    })
```

Comment:
This is not a very efficient way to calculate the closest point. You can get it in O(1):
https://github.com/concord-consortium/tectonic-explorer/blob/master/js/plates-model/grid.js#L94-L101
(once you have field ID, you can have field and it's absolute position too)
You should try to read the code of the whole class (it's not too long), and see how it's used in various places. It's core of the model grid and many many calculations, so it's very helpful to understand how it works. Basically, this helper lets you find the nearest field to a given position in O(1). This is the main thing used in the collision detections.

I wouldn't care about "colliding" property either, but I'll talk about it more in another comment.